lib/api/controllers/authController.js

```diff
@@ -163,10 +196,109 @@ class AuthController {
   /**
    * List authentication strategies
    *
-   * @returns {Promise}
+   * @returns {Promise.<string[]>}
    */
   getStrategies() {
-    return this.kuzzle.pluginsManager.listStrategies();
+    return Bluebird.resolve(this.kuzzle.pluginsManager.listStrategies());
+  }
+
+  /**
+   * @param {Request} request
+   * @returns {Promise.<Object>}
+   */
+  createMyCredentials(request) {
+    const
+      createMethod = this.kuzzle.pluginsManager.getStrategyMethod(request.input.args.strategy, 'create'),
+      validateMethod = this.kuzzle.pluginsManager.getStrategyMethod(request.input.args.strategy, 'validate');
+
+    assertIsConnected(this.kuzzle, request);
```

Comment:
nipticking: assertions checks should be done at the top of the function, to avoid unnecessary calls on `getStrategyMethod`
(same remark for other methods)